### PR TITLE
Use GitHub reusable workflows

### PR DESCRIPTION
**Pull request by davidtaylorhq**

### CODE REVIEW

**Comment by davidtaylorhq** [comment]
Demonstrations in:
- https://github.com/discourse/discourse-assign/pull/412
- https://github.com/discourse/discourse-prometheus-alert-receiver/pull/59

File: .github/workflows/discourse-plugin.yml [hunk 1]
```diff
@@ -1,17 +1,77 @@
...
+          node-version: 18
+          cache: yarn
+
+      - name: Yarn install
+        run: yarn install
+
+      - name: Set up ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.7
```
Comment (CvX) [inline .github/workflows/discourse-plugin.yml:33]: Let's gooooo 😄 

```suggestion
          ruby-version: '3.1'
```

File: actions/clone-additional-plugins/action.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+name: Install Additional Plugins
+description: Installs any additional plugins listed in this plugin/theme's `about.json` (under `.tests.requiredPlugins`)
+inputs:
+  ssh_private_key:
+    description: 'Optional SSH private key, if one or more of the repos is private'
+    required: false
+runs:
+  using: "composite"
+  steps:
+    - name: Configure SSH Keys
```
Comment (CvX) [inline actions/clone-additional-plugins/action.yml:10]: Just to align with workflow template formatting 😃 

```suggestion

inputs:
  ssh_private_key:
    description: 'Optional SSH private key, if one or more of the repos is private'
    required: false

runs:
  using: "composite"

  steps:
    - name: Configure SSH Keys
```

**Review by CvX — APPROVED** [review]
🧙🚀

**Pull request merged**